### ISSUE TITLE
[bug] DHCP server + Unbound DNS = Client registration is intermittent and unreliable

### ISSUE THREAD

**Issue by steakknife** [opnsense/core#6982]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [x] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [x] I am convinced that my issue is new after having checked both open and closed issues at https://github.com/opnsense/core/issues?q=is%3Aissue

\>\> Potentially related #3478 <<

**Describe the bug**

1. New hosts do not self-register their reported hostnames with unbound via DHCP.

**To Reproduce**

(More fs or configuration state is likely necessary to reproduce this behavior.)

1. Flush caches on macOS network admin client using `sudo dscacheutil -flushcache; sudo killall -HUP mDNSResponder; sudo arp -d -a`
2. Bring a new client (new hostname + new MAC address) IPv4 DHCP or dual-stack host online without mDNS/Bonjour/avahi that reports its hostname to DHCP
3. `while :; do dig @{{router}} {{hostname}}; sleep 5; done` never resolves

**Expected behavior**

1. Name resolution works within about 1 minute

**Describe alternatives you considered**

1. Restart DHCPd
2. Restart Unbound\* 
3. Reboot OPNsense\*
4. Review OPNsense support forums for similar issues
5. Review OPNsense github.com issues
6. Review FreeBSD/Unbound bugs

\* Which sometimes works, but this is not a scalable approach.

**Screenshots**

N/A

**Relevant log files**

Redacted logs and config available upon request.

**Additional context**

This feature worked correctly previously. Changes between then and now were firmware upgrades and dual-stack IPv6. It'd be nice to have dual-stack and IPv6-only congruence because more environments are moving to IPv6 only for new environments.

**Environment**

- hw: Deciso 740 v1
- os: OPNsense Business 23.4.2_4
- services enabled and running: configd, cron, dhcpd, login, miniupnpd, ntpd, openssh, pf, radvd, routing, sysctl, syslog-ng, unbound, webgui, wireguard
- unbound
	- upstream dns: DoH to IPv4 and IPv6 dns.google (ISP)
	- DNS and DNSSEC are working over IPv4 and IPv6
	- internal domain for all hosts: .lan
- network
	- l2
		- WAN: 10GBase-SFI to ISP
		- LAN: 10GBase-SFI to distribution switch
	- l3
		- WAN: DHCP public IPv4 & DHCPv6 /56
		- LAN: Static IPv4 & Track interface (IPv6)
		- Public web performance and connectivity verified over both IPv4 and IPv6.

**Comment by steakknife** [comment]
Same results with 23.10. I need an update on this because it's nonfunctional and a blocking item.

**Comment by steakknife** [comment]
Another related, unresolved problem without an answer: #4506

**Comment by AdSchellevis** [comment]
The watcher process should send messages to the log, it just seems to appear in the wrong one (Services: Captive Portal: Log File). https://github.com/opnsense/core/commit/2a58142ddbd0383a7897dbf09797339f4e2d6ee5 will fix the location, in order to debug your setup it's best to first check the dhcp messages in the log as these show registrations and expired leases. 

Messages emitted to the log include "dhcpd entry changed ..." when a new lease is registered and "dhcpd expired..." when one has expired. 

Often it helps to peel of complexity to find issues in your setup, combinations of services might sometimes react differently than expected.

**Comment by steakknife** [comment]
### tl;dr: The box is in a state that can no longer repro and so there's no MCE.

Thanks for following up.

Murphy's law of nondeterministic faults: of course it works now.

I take great pains to simplify things because `simplicity α reliability` (usually) with experimental validation.

After several reboots and `/usr/local/sbin/pluginctl -c unbound_start` (how the cron does it), DHCPd registers in Unbound < 500ms.

## Positive verification procedure

1. On 2 observer LAN hosts `while :; do dig -4 @{{opnsense-lan-ip}} {{novel-hostname}} | grep 'A[[:space:]]*[0-9]' || echo 'didnt resolve'; sleep 1; done`.

2. On a victim LAN box (CentOS 9 Stream), `hostnamectl set-hostname {{novel-hostname}} && systemctl restart NetworkManager`. 

3. The observers should report `{{novel-hostname}}`

## Future remediation

1. GDrive config backups were already occurring frequently.

2. I setup periodic zfs snapshot sends to a plain FreeBSD 13 VM box for future investigation should disk-persistent state cause it to reoccur.
